File: uniworld-frontend/src/app/interfaces/Playlist.ts
import type { Song } from './Song';
import type { User } from './User';

export interface Playlist {
	playlistID: number;
	name: string;
	isPublic: boolean;
	coverImage: string | null;
	user: User;
	songs: Song[];
}

export interface PlaylistRequest {
	name: string;
	isPublic?: boolean;
	coverImage?: string | null;
	songIds?: number[];
}

export interface PlaylistResponse {
	playlistID: number;
	name: string;
	isPublic: boolean;
	coverImage: string | null;
	user: User;
	songs: Song[];
}

File: uniworld-backend/src/main/java/com/uniworld/uniworld_backend/Playlist.java
package com.uniworld.uniworld_backend;

import java.util.List;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.NoArgsConstructor;
import lombok.ToString;

@Entity
@Table(name = "playlists")
@Data
@NoArgsConstructor
@AllArgsConstructor
public class Playlist {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long playlistID;

    @Column(nullable = false)
    private String name;

    @Column(name="is_public", nullable = false)
    private Boolean isPublic;

    @Column(name="cover_image")
    private String coverImage;

    @ManyToOne(fetch = FetchType.EAGER)
    @JoinColumn(name = "user_id", nullable = false)
    @ToString.Exclude
    @EqualsAndHashCode.Exclude
    private User user;

    @ManyToMany(fetch = FetchType.EAGER)
    @JoinTable(
        name = "playlist_songs",
        joinColumns = @JoinColumn(name = "playlist_id"),
        inverseJoinColumns = @JoinColumn(name = "song_id")
    )
    @ToString.Exclude
    @EqualsAndHashCode.Exclude
    private List<Song> songs;
}
